packages/protocol/src/abstracts/BaseRouter.sol

```diff
@@ -431,13 +473,14 @@ abstract contract BaseRouter is SystemAccessControl, IRouter {
    *
    * @param user address to verify is the beneficiary
    */
-  function _checkBeneficiary(address user) internal {
-    if (_beneficiary == address(0)) {
-      _beneficiary = user;
+  function _checkBeneficiary(address beneficiary, address user) internal pure returns (address) {
+    if (beneficiary == address(0)) {
+      return user;
     } else {
-      if (_beneficiary != user) {
+      if (beneficiary != user) {
         revert BaseRouter__bundleInternal_notBeneficiary();
       }
+      return user;
     }
```

Comment:
We can avoid the nested condition so it can be re-written with "if - else if - else".